kernel/src/main/scala/cats/kernel/Group.scala

```diff
@@ -0,0 +1,48 @@
+package cats.kernel
+
+import scala.{ specialized => sp }
+
+/**
+ * A group is a monoid where each element has an inverse.
+ */
+trait Group[@sp(Int, Long, Float, Double) A] extends Any with Monoid[A] {
+
+  /**
+   * Find the inverse of `a`.
+   *
+   * `combine(a, inverse(a))` = `combine(inverse(a), a)` = `empty`.
+   */
+  def inverse(a: A): A
+
+  /**
+   * Remove the element `b` from `a`.
+   *
+   * Equivalent to `combine(a, inverse(a))`
+   */
+  def remove(a: A, b: A): A = combine(a, inverse(b))
+
+  /**
+   * Return `a` appended to itself `n` times. If `n` is negative, then
+   * this returns `inverse(a)` appended to itself `n` times.
+   */
+  override def combineN(a: A, n: Int): A =
+    if (n > 0) repeatedCombineN(a, n)
+    else if (n == 0) empty
+    else if (n == Int.MinValue) combineN(inverse(combine(a, a)), 1073741824)
```

Comment:
this is due to:
1. `- Int.MinValue == Int.MinValue`
2. `combineN(x, n) == combineN(inverse(x), -n)`
3. `combineN(x, n*n) = combineN(combine(x, x), n)`

So, `1073741824 == -(Int.MaxValue / 2)`

Can we add some verbosity to the comments?